applications/sink/rsocket-sink/src/test/java/org/springframework/cloud/stream/app/rsocket/sink/test/RsocketSinkTestApplication.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.rsocket.sink.test;
+
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.fn.consumer.rsocket.RsocketConsumerConfiguration;
+import org.springframework.context.annotation.Import;
+
+@SpringBootApplication
```

Comment:
You probably still can keep this class as inner one without conflicts with that `RSocketserverApplication` if you also use `@EnableAutoConfiguration` and `@SpringBootConfiguration` instead of this `@SpringBootApplication`.
I think you also would need to add `classes` into the `@SpringBootTest` to avoid another conflict.

Although this is IMHO for design how to hide infrastructure of one test from another...